htdocs/includes/db_lib.php

```diff
@@ -5817,9 +5817,21 @@ function add_user($user)
 		if($user->level == 17) {
 			$user->rwoptions = LabConfig::getDoctorUserOptions();
 		}
+		// If level = 20, generate the satellite_lab_id value
+        if ($user->level == 20) {
+            // Retrieve the last satellite_lab_id (assuming it's auto-incremented)
+            $query_string = "SELECT MAX(satellite_lab_id) AS max_lab_id FROM user";
+            $result = mysql_query($query_string);
+            $row = mysql_fetch_assoc($result);
+            $new_satellite_lab_id = $row['max_lab_id'] + 1;
```

Comment:
Clever! So each user gets their own satellite lab ID, and (if you only use the UI) you can't have two users for the same satellite lab?

That makes sense, we'll just have to make sure to document it.